This diff was generated against enlightenment-0.25.0, targeting all the occurances of 'suid_exes'.

diff --git a/src/bin/e_auth.c b/src/bin/e_auth.c
index 6d07a0ac3..31e0e728f 100644
--- a/src/bin/e_auth.c
+++ b/src/bin/e_auth.c
@@ -38,8 +38,7 @@ e_auth_begin(char *passwd)
    pwlen = strlen(passwd);
 
    snprintf(buf, sizeof(buf),
-            "%s/enlightenment/utils/enlightenment_ckpasswd pw",
-            e_prefix_lib_get());
+            "/run/setuid-programs/enlightenment_ckpasswd pw");
    exe = ecore_exe_pipe_run(buf, ECORE_EXE_PIPE_WRITE, NULL);
    if (!exe) goto out;
    snprintf(buf, sizeof(buf), "pw %s", passwd);
@@ -75,8 +74,7 @@ e_auth_polkit_begin(char *passwd, const char *cookie, unsigned int uid)
    pwlen = strlen(passwd);
 
    snprintf(buf, sizeof(buf),
-            "%s/enlightenment/utils/enlightenment_ckpasswd pk",
-            e_prefix_lib_get());
+            "/run/setuid-programs/enlightenment_ckpasswd pk");
    exe = ecore_exe_pipe_run(buf, ECORE_EXE_PIPE_WRITE, NULL);
    if (!exe) goto out;
    snprintf(buf, sizeof(buf), "%s %u %s", cookie, uid, passwd);
diff --git a/src/bin/e_fm/e_fm_main_eeze.c b/src/bin/e_fm/e_fm_main_eeze.c
index 9b10b3117..74e6b72ad 100644
--- a/src/bin/e_fm/e_fm_main_eeze.c
+++ b/src/bin/e_fm/e_fm_main_eeze.c
@@ -318,7 +318,7 @@ _e_fm_main_eeze_volume_eject(E_Volume *v)
      {
         char buf[PATH_MAX];
 
-        snprintf(buf, sizeof(buf), "%s/enlightenment/utils/enlightenment_sys", eina_prefix_lib_get(pfx));
+        snprintf(buf, sizeof(buf), "/run/setuid-programs/enlightenment_sys");
         eeze_disk_mount_wrapper_set(v->disk, buf);
      }
    v->guard = ecore_timer_loop_add(E_FM_EJECT_TIMEOUT, (Ecore_Task_Cb)_e_fm_main_eeze_vol_eject_timeout, v);
@@ -512,7 +512,7 @@ _e_fm_main_eeze_volume_unmount(E_Volume *v)
      {
         char buf[PATH_MAX];
 
-        snprintf(buf, sizeof(buf), "%s/enlightenment/utils/enlightenment_sys", eina_prefix_lib_get(pfx));
+        snprintf(buf, sizeof(buf), "/run/setuid-programs/enlightenment_sys");
         eeze_disk_mount_wrapper_set(v->disk, buf);
      }
    v->guard = ecore_timer_loop_add(E_FM_UNMOUNT_TIMEOUT, (Ecore_Task_Cb)_e_fm_main_eeze_vol_unmount_timeout, v);
@@ -548,7 +548,7 @@ _e_fm_main_eeze_volume_mount(E_Volume *v)
      {
         char buf2[PATH_MAX];
 
-        snprintf(buf2, sizeof(buf2), "%s/enlightenment/utils/enlightenment_sys", eina_prefix_lib_get(pfx));
+        snprintf(buf2, sizeof(buf2), "/run/setuid-programs/enlightenment_sys");
         eeze_disk_mount_wrapper_set(v->disk, buf2);
      }
    v->guard = ecore_timer_loop_add(E_FM_MOUNT_TIMEOUT, (Ecore_Task_Cb)_e_fm_main_eeze_vol_mount_timeout, v);
diff --git a/src/bin/e_start_main.c b/src/bin/e_start_main.c
index 722063339..2da2d5a23 100644
--- a/src/bin/e_start_main.c
+++ b/src/bin/e_start_main.c
@@ -596,8 +596,7 @@ main(int argc, char **argv)
               eina_prefix_data_get(pfx));
    putenv(buf2);
    myasprintf(&buf3,
-              "E_ALERT_SYSTEM_BIN=%s/enlightenment/utils/enlightenment_system",
-              eina_prefix_lib_get(pfx));
+              "E_ALERT_SYSTEM_BIN=/run/setuid-programs/enlightenment_system");
    putenv(buf3);
 
    home = getenv("HOME");
diff --git a/src/bin/e_system.c b/src/bin/e_system.c
index bfd43e7e2..45d78b7ed 100644
--- a/src/bin/e_system.c
+++ b/src/bin/e_system.c
@@ -133,7 +133,7 @@ _system_spawn(void)
    else _respawn_count = 0;
    if (_respawn_count > 5) return;
    snprintf(buf, sizeof(buf),
-            "%s/enlightenment/utils/enlightenment_system", e_prefix_lib_get());
+            "/run/setuid-programs/enlightenment_system");
    _system_exe = ecore_exe_pipe_run
      (buf, ECORE_EXE_NOT_LEADER | ECORE_EXE_TERM_WITH_PARENT |
       ECORE_EXE_PIPE_READ | ECORE_EXE_PIPE_WRITE, NULL);
